sdk/appconfiguration/app-configuration/sample-react/src/components/Page.tsx

```diff
@@ -57,18 +57,30 @@ export default function Page(): JSX.Element {
       )
     );
     if (isFeatureFlag(setting1)) {
-      // console.log(`${setting1.key} is enabled : ${setting1.enabled}`, setting1);
-      setFeature1({ enabled: setting1.enabled });
+      const parsedFeatureFlag1 = parseFeatureFlag(setting1);
+      // console.log(`${parsedFeatureFlag1.key} is enabled : ${parsedFeatureFlag1.enabled}`, parsedFeatureFlag1);
+      setFeature1({ enabled: parsedFeatureFlag1.value.enabled });
     }
     if (isFeatureFlag(setting2)) {
-      // console.log(`${setting2.key} is enabled : ${setting2.enabled}`, setting2);
-      const clientFilter = setting2.conditions.clientFilters?.[0];
-      if (isFeatureFlagClientFilter("timeWindow", clientFilter)) {
-        const now = Date.now();
-        const withinRange =
-          now - Date.parse(clientFilter.parameters.start) > 0 &&
-          Date.parse(clientFilter.parameters.end) - now > 0;
-        setFeature2({ enabled: withinRange });
+      const parsedFeatureFlag2 = parseFeatureFlag(setting2);
+      // console.log(`${parsedFeatureFlag2.key} is enabled : ${parsedFeatureFlag2.value.enabled}`, parsedFeatureFlag2);
+      const clientFilter = parsedFeatureFlag2.value.conditions.clientFilters?.[0];
+      if (clientFilter.name === "Microsoft.TimeWindow") {
+        if (
+          clientFilter.parameters &&
+          clientFilter.parameters["Start"] &&
+          clientFilter.parameters["End"] &&
+          typeof clientFilter.parameters["Start"] === "string" &&
+          typeof clientFilter.parameters["End"] === "string"
```

Comment:
recommendation: make a User-Defined Typeguard

Reply:
Done. Let me know how that looks!